### PR TITLE
add interface to set minimal gas price

### PR DESCRIPTION
**Pull request by HarryLiIsMe**
* **make sure that you have executed all the following process and no errors occur**
  - [ ] make fmt
  - [ ] make lint
  - [ ] make test

* **The major changes of this PR**

* **The major impacts of this PR**
  - [ ] Impact WASM?
  - [ ] Impact Web3 API?
  - [ ] Impact mainnet data compatibility?

* **Extra documentations**

### CODE REVIEW

File: src/components/contracts/rpc/src/eth.rs [hunk 1]
```diff
@@ -190,11 +191,27 @@ impl EthApi for EthApiImpl {
             Err(e) => return Box::pin(future::err(e)),
         };
 
+        let ctx = match self
```
Comment (ws4charlie) [inline src/components/contracts/rpc/src/eth.rs:194]: I think there is a `fn block_number()` function to get current height for convenience.

File: src/components/config/src/abci/mod.rs [hunk 2]
```diff
@@ -36,6 +36,8 @@ pub struct CheckPointConfig {
     pub utxo_asset_prefix_height: u64,
     pub prism_bridge_address: String,
     pub nonce_bug_fix_height: u64,
+    pub proper_gas_set_height: u64,
+    pub proper_gas_set_value: u64,
```
Comment (ws4charlie) [inline src/components/config/src/abci/mod.rs:40]: This `proper_gas_set_value` doesn't need to appear in checkpoint. I think adding only `proper_gas_set_height` will do the job.

File: src/components/contracts/modules/ethereum/src/lib.rs [hunk 3]
```diff
@@ -202,11 +202,17 @@ impl<C: Config> ValidateUnsigned for App<C> {
             )));
         }
 
-        if transaction.gas_price < C::FeeCalculator::min_gas_price() {
+        let min_gas_price =
+            if ctx.header.height as u64 > CFG.checkpoint.proper_gas_set_height {
```
Comment (ws4charlie) [inline src/components/contracts/modules/ethereum/src/lib.rs:206]: In stead of distributing fee calculation logic in many places, can we reorganize the fee calculation forking, by height, logic in one single place like below.
image

File: src/components/contracts/rpc/src/eth.rs [hunk 4]
```diff
@@ -373,7 +390,19 @@ impl EthApi for EthApiImpl {
     }
 
     fn gas_price(&self) -> Result<U256> {
-        Ok(<BaseApp as module_evm::Config>::FeeCalculator::min_gas_price())
+        let ctx = self
+            .account_base_app
```
Comment (ws4charlie) [inline src/components/contracts/rpc/src/eth.rs:394]: Same `fn block_number()` func will help.

Comment (HarryLiIsMe) [reply src/components/contracts/rpc/src/eth.rs]: Ok, I have replaced it with this function.

File: src/components/contracts/rpc/src/eth.rs (hunk 1 above)
Comment (HarryLiIsMe) [reply src/components/contracts/rpc/src/eth.rs]: Ok, I have replaced it with this function.

**Review by ws4charlie — APPROVED** [review]
looks good

**Pull request closed without merging**